packages/models/src/huggingface/text-generation.ts

```diff
@@ -0,0 +1,210 @@
+import { HttpError, POST } from '@axflow/models/shared';
+
+// HuggingFace has the concept of a task. This code supports the "textGeneration" task.
+// https://huggingface.co/models?pipeline_tag=text-generation
+
+// https://huggingface.co/docs/api-inference/quicktour#running-inference-with-api-requests
+const HF_MODEL_API_URL = 'https://api-inference.huggingface.co/models/';
+
+function headers(accessToken?: string) {
+  const headers: Record<string, string> = {
+    accept: 'application/json',
+    'content-type': 'application/json',
+  };
+  if (typeof accessToken === 'string') {
+    headers.authorization = `Bearer ${accessToken}`;
+  }
+  return headers;
+}
+
+export namespace HfChatTypes {
+  // https://huggingface.co/docs/api-inference/detailed_parameters#text-generation-task
+  export type Request = {
+    model: string;
+    stream?: boolean;
```

Comment:
Let's remove this from the request params. While this is a param to the underlying API, our methods handle adding this value, not the user, and this type is user-facing.

Reply:
Ah, good call, we override it anyway. Will remove